fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1543,8 +1543,6 @@ def _register_post_backward_hooks(self) -> None:
             return  # don't register grad hooks if grad isn't enabled
         for p in self.params:
             if p.requires_grad:
-                if hasattr(p, "_shard_bwd_hook"):
```

Comment:
Do the comments above need to updated because it seems like we don't need to remove the hook at the end of the BW?

Reply:
ohh yeah definitely, will do that

Reply:
if the two lines are removed, will the hooks fire multiple times for multiple forwards cases (e.g. multiple activation checkpointing)?

Reply:
even for single activation checkpointing case like Checkpoint(FSDP(module)), if there is forward recomputation in the backward pass, the hooks will be registered twice and fired twice unexpectedly?